src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
Drew and I had a hallway conversation.  Originally, there was an intent of having every object type foo have a matching options::foo type.  If we want to stick with that, we could have `options::pool` and have that constructable from `options::client`.  Let's put this on our list to discuss tomorrow.